### PR TITLE
Support time range in promtool query labels

### PR DESCRIPTION
**Pull request by yeya24**
Signed-off-by: yeya24 <yb532204897@gmail.com>

This pr adds `start` and `end` flags in `promtool query labels` command to support time range query.

### CODE REVIEW

File: cmd/promtool/main.go [hunk 1]
```diff
@@ -549,10 +551,35 @@ func QueryLabels(url *url.URL, name string, p printer) int {
 		return 1
 	}
 
+	var (
+		minTime = time.Now().Add(-9999 * time.Hour)
+		maxTime = time.Now().Add(9999 * time.Hour)
+	)
+
+	var stime, etime time.Time
```
Comment (bwplotka) [inline cmd/promtool/main.go:559]: you can merge those `vars` (:

File: cmd/promtool/main.go [hunk 2]
```diff
@@ -549,10 +551,35 @@ func QueryLabels(url *url.URL, name string, p printer) int {
...
+
+	if start == "" {
+		stime = minTime
```
Comment (bwplotka) [inline cmd/promtool/main.go:562]: what about 
```
stime := minTime
if start != "" {
// change stime
}
```
and same below?

File: go.mod [hunk 3]
```diff
@@ -38,7 +38,7 @@ require (
 	github.com/hashicorp/serf v0.9.0 // indirect
 	github.com/influxdata/influxdb v1.8.0
 	github.com/jpillora/backoff v1.0.0 // indirect
-	github.com/json-iterator/go v1.1.9
+	github.com/json-iterator/go v1.1.10
```
Comment (bwplotka) [inline go.mod:41]: ideally we upgrade those in separate PR. Or at least you can mention in commit `update client_golang` (:

Comment (yeya24) [reply go.mod:46]: Changelog added.

File: cmd/promtool/main.go (hunk 2 above)
Comment (yeya24) [reply cmd/promtool/main.go]: Thanks, I like this way. Much cleaner than before.

File: go.mod (hunk 3 above)
Comment (beorn7) [reply go.mod:46]: Hmm, I don't think we should mention it in the CHANGELOG. I think @bwplotka was referring to the commit description. Furthermore, this is about json-iterator, not client_golang. But I guess it was _triggered_ by the client_golang update (which is normal with the way Go modules work).

So I'd say just add "Updated prometheus/client_golang and json-iterator/go" to the commit description (not the changelog).

Comment (bwplotka) [reply go.mod:46]: Yes, I meant commit message (:

Comment (yeya24) [reply go.mod:46]: Sorry I misunderstood :(

PR updated, PTAL.

**Pull request merged**